api/v01/entities/vrp_input.rb

```diff
@@ -0,0 +1,521 @@
+# Copyright © Mapotempo, 2020
+#
+# This file is part of Mapotempo.
+#
+# Mapotempo is free software. You can redistribute it and/or
+# modify since you respect the terms of the GNU Affero General
+# Public License as published by the Free Software Foundation,
+# either version 3 of the License, or (at your option) any later version.
+#
+# Mapotempo is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+# or FITNESS FOR A PARTICULAR PURPOSE.  See the Licenses for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Mapotempo. If not, see:
+# <http://www.gnu.org/licenses/agpl.html>
+#
+require './api/v01/vrp'
+
+module VrpInput
+  extend Grape::API::Helpers
+
+  params :input do
+    optional(:vrp, type: Hash, documentation: { param_type: 'body' }, coerce_with: ->(c) { c.has_key?('filename') ? JSON.parse(c[:tempfile].read) : c }) do
+      use :request_object
+    end
+    use :request_files
+    exactly_one_of :vrp, :vehicles # either it is a json (and :vrp is required) or it is a csv (and :vehicles is required)
+  end
+
+  # Input as expected in JSON
+  params :request_object do
+    optional(:name, type: String, desc: 'Name of the problem, used as tag for all element in order to name plan when importing returned .csv file')
+    optional(:matrices, type: Array, documentation: { desc: 'Define all the distances between each point of problem' }) do use :vrp_request_matrices end
+
+    optional(:points, type: Array, documentation: { desc: 'Particular place in the map' }) do use :vrp_request_point end
+
+    optional(:units, type: Array, documentation: { desc: 'The name of a Capacity/Quantity' }) do use :vrp_request_unit end
+
+    optional(:rests, type: Array, documentation: { desc: 'Break within a vehicle tour' }) do use :vrp_request_rest end
+
+    optional(:zones, type: Array, desc: '') do use :vrp_request_zone end
+
+    requires(:vehicles, type: Array, documentation: { desc: 'Usually represent a work day of a particular driver/vehicle' }) do
```

Comment:
Could we eventually change this ? This could be confusing for people using `first_solution_strategy = periodic`

```suggestion
    requires(:vehicles, type: Array, documentation: { desc: 'Set of available vehicles.' }) do
```